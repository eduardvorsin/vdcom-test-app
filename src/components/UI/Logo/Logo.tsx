import React from 'react';
import {
  Link,
  SxProps,
  Theme,
  styled,
} from '@mui/material';
import { Link as RouterLink } from 'react-router-dom';
import Logotype from '../../../assets/images/icons/logo.svg?url';

type LogoProps = {
  sx?: SxProps<Theme>,
};

const StyledLogo = styled('img')(({ theme }) => ({
  display: 'block',
  width: '100px',
  height: '27px',
  [theme.breakpoints.up('sm')]: {
    width: '110px',
    height: '30px',
  },
  [theme.breakpoints.up('md')]: {
    width: '140px',
    height: '39px',
  },
  [theme.breakpoints.up('lg')]: {
    width: '175px',
    height: '48px',
  },
}));

const Logo: React.FC<LogoProps> = ({
  sx,
}) => (
  <Link
    to='/contacts'
    component={RouterLink}
    sx={{
      display: 'inline-flex',
      ...sx,
    }}
  >
    <StyledLogo
      src={Logotype}
      alt='logo'
    />
  </Link>
);

export default Logo;
