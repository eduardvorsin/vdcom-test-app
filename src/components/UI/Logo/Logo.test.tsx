import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import Logo from './Logo';

describe('Logo tests', () => {
  it('should render correctly', () => {
    render(
      <MemoryRouter initialEntries={['/contacts']}>
        <Routes>
          <Route path='contacts' element={<Logo className='logo' />} />
        </Routes>
      </MemoryRouter>,
    );

    expect(screen.getByTestId<HTMLAnchorElement>('logo-link')).toBeInTheDocument();
    expect(screen.getByAltText<HTMLImageElement>('logo')).toBeInTheDocument();
  });

  it('is a basic snapshot', () => {
    render(
      <MemoryRouter initialEntries={['/contacts']}>
        <Routes>
          <Route path='contacts' element={<Logo className='logo' />} />
        </Routes>
      </MemoryRouter>,
    );

    expect(screen.getByTestId<HTMLAnchorElement>('logo-link')).toMatchSnapshot();
  });
});
