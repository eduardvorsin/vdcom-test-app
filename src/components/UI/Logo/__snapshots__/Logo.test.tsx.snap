// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo tests is a basic snapshot 1`] = `
<a
  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways logo css-h74647-MuiTypography-root-MuiLink-root"
  data-testid="logo-link"
  href="/contacts"
>
  <img
    alt="logo"
    class="css-12pywbb"
    src="assets"
  />
</a>
`;
